### PR TITLE
Landing Page

### PR DESCRIPTION
**Pull request by weepotty**
# Description

This is the landing page. As far as possible I've tried to use pre-existing CSS classes.

Fixes # (issue)
https://3.basecamp.com/5349022/buckets/32784806/todos/6162939797

## Type of change
- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?
- On desktop
Screenshot 2023-05-22 at 9 10 29 AM
Screenshot 2023-05-22 at 9 10 39 AM

- On mobile screen
Screenshot 2023-05-22 at 9 10 50 AM

Screenshot 2023-05-22 at 9 10 44 AM

# Checklist:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added proof(eg. screenshots) that prove my fix is effective or that my feature works

### CODE REVIEW

File: world/index.html [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+layout: world
+permalink: /world
+---
+
+<div class="landing">
+  <section class="landing__top common-background--white">
+    <figure
+      class="common-background--grey common-shape--bottom-grey-down-left"
+    ></figure>
+    <a href="#skyline"
```
Comment (ayushn21) [inline world/index.html:11]: This isn't a hard and fast rule, but I think putting the `>` for a tag on a new line is less readable. I prefer this formatting:

```
<a href="#skyline">
  <img src="../assets/images/logo-railsworld.svg" alt="rails world logo" />
</a>
```

This is personal preference so this is purely advisory, but I'm curious about what you think? For consistency with the rest of the project, I reckon it's worth changing though.

If you do change it, please change it everywhere in the file.

File: world/index.html [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+      ><img src="../assets/images/logo-railsworld.svg" alt="rails world logo"
```
Comment (ayushn21) [inline world/index.html:12]: Another purely advisory ... I shouldn't even be mentioning this it's so unbelievably minor but my OCD is making me at least leave a comment :D 

I think we should capitalise the alt text: "Rails World Logo".

(One hell of a nitpick, I know, sorry!)

File: world/index.html [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    /></a>
+  </section>
+  <section class="landing__top common-background--grey">
+    <ul class="landing__ul common-padding--bottom-small">
+      <li class="common-content"><strong>Dates:</strong> 5 & 6 Oct 2023</li>
+      <li class="common-content">
+        <strong>Location:</strong> Beurs van Berlage, Amsterdam, NL
```
Comment (ayushn21) [inline world/index.html:19]: I wonder if it's worth linking this to Google Maps so folks can conveniently see where the venue is?

File: _sass/modules/_world.scss [hunk 4]
```diff
@@ -0,0 +1,63 @@
+.landing {
+  text-align: center;
+
+  &__top {
+  height: 50vh; // for browsers that don't support svh
+  height: 50svh;
+  position:relative;
+  display: flex;
+  justify-content: center;
+  align-items:end;
```
Comment (ayushn21) [inline _sass/modules/_world.scss:10]: Please ensure there's 1 space between the key and value of the style rule for consistency :)

File: world/index.html [hunk 5]
```diff
@@ -0,0 +1,37 @@
+---
+layout: world
+permalink: /world
+---
+
+<div class="landing">
+  <section class="landing__top common-background--white">
```
Comment (ayushn21) [inline world/index.html:7]: Love the use of `<section>` rather than `<div>` everywhere on this page, good choice!

File: world/index.html [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+  </section>
+  <section class="landing__top common-background--grey">
+    <ul class="landing__ul common-padding--bottom-small">
+      <li class="common-content"><strong>Dates:</strong> 5 & 6 Oct 2023</li>
+      <li class="common-content">
+        <strong>Location:</strong> Beurs van Berlage, Amsterdam, NL
+      </li>
+      <a
+        class="common-button landing__button common-button--version common-button--color-red"
+        href="https://www.papercall.io/rails-world-2023"
```
Comment (ayushn21) [inline world/index.html:23]: Let's add `target="_blank"` here so the link opens in a new tab.

**Review by ayushn21 — CHANGES_REQUESTED** [review]
Fantastic stuff @weepotty, seriously good job! 

I've left a handful of comments, some purely advisory but I think we're 90% of the way there. I'll take another look tomorrow because I'm half asleep so there's every change I've missed something 😂 

Before we merge, it's worth cleaning up the commits so the commit history is always sequential and logical. Here's how I usually do this on other projects:

```shell
$ git fetch
$ git reset --mixed origin/world
# This will reset the state of your current branch to origin/world and all your changes will be in the working copy
# Re-commit your changes in a logical order. For eg: Create world layout, Add amsterdam skyline SVG, Create world landing page ..... something along those lines .... that way, someone reading the commit history has a clear picture of the sequence of changes and there are no "work in progress" commits. Make sense?

$ git push --force
```

The above is a destructive change and hence risky ... if you're unsure about what you're doing, ping me, happy to virtually "look over your shoulder" while you do it to ensure you don't lose your work.

**Comment by weepotty** [comment]
thanks 🙏🏻 btw this PR preview feature is great!

**Comment by weepotty** [comment]
Thanks for all the feedback Ayush, I've made the changes suggested - you can have a look at the code and let me know if you spot anything else. Whenever we are happy to merge, I'll clean up the commits.

**Review by ayushn21 — APPROVED** [review]
Awesome ... looks good to me 👏  .... @AmandaPerino please take a peek if you've got any further feedback ... from my end this is good to go ... I'll merge it once you clean the commit history @weepotty 👍

**Comment by AmandaPerino** [comment]
> Awesome ... looks good to me 👏 .... @AmandaPerino please take a peek if you've got any further feedback ... from my end this is good to go ... I'll merge it once you clean the commit history @weepotty 👍

Looks great! Go for it!

**Comment by AmandaPerino** [comment]
Are you all okay with the landing page **not** having the disclaimer? It should definitely be on the full website. 

"This website was created by <Dev Name & Link> & <Dev Name & Link> , junior developers from the Rails community.
It was commissioned by the Rails Foundation, designed by <Designer Name & Link>,  and created under the mentorship of <Mentor Name & Link>."

**Comment by ayushn21** [comment]
I'm not fussed about it for the landing page since it's temporary, but happy to have it in there if @weepotty and Mohamed would prefer.

**Comment by AmandaPerino** [comment]
Sorry one more thing - can you please add a banner to the homepage which links to /world with the text: **Join us for the first Rails World this October in Amsterdam. Tickets on sale soon.** If a banner is too much to add right now, perhaps just an **Events** tab in the navbar?

**Comment by weepotty** [comment]
> I reckon a banner is pretty straightforward ... best not to upset the balance of the navbar in my opinion ... also the navbar is hidden away in a burger menu on mobile, whereas a banner will always be visible so definitely the way to go in my opinion!
> 
> I'd probably put it right at the top of the page, above the navbar, flush with the top of the viewport ... What do you think?
> 
> @weepotty do you want to take a stab at it? Please do it in a separate PR though.

Sure I'll have a go at the banner tomorrow.

**Comment by AmandaPerino** [comment]
@weepotty #DBD3D5 (the pale pink) with black text. To match this box please:
Screenshot 2023-05-23 at 16 35 24

@ayushn21 Are we setting the landing page live today already? I'd like to update the board.

**Comment by ayushn21** [comment]
@weepotty I think you should be able to re-use the existing `common-button` and `common-button--version` classes to match the button styling.

@AmandaPerino Nope, I thought we could go live once we have the banner and social image ready as well. I'll merge this PR once we fix the button but that won't push it live .... we'll need to create a PR into the main repository for that which I think is worth doing once we have everything in place. Is that alright?

**Pull request merged**